support-models/src/main/scala/com/gu/support/workers/states/FailureHandlerState.scala

```diff
@@ -12,11 +12,19 @@ case class FailureHandlerState(
   user: User,
   giftRecipient: Option[GiftRecipient],
   product: ProductType,
-  paymentFields: Option[Either[PaymentFields, RedemptionData]], //Will be present if CreatePaymentMethod failed
-  paymentMethod: Option[Either[PaymentMethod, RedemptionData]], //Will be present if anything after CreatePaymentMethod failed
+  paymentProvider: PaymentProvider,
   firstDeliveryDate: Option[LocalDate],
   promoCode: Option[PromoCode]
-) extends StepFunctionUserState
+) extends FailableState
+
+trait FailableState extends StepFunctionUserState {
+  // only required fields needed here
+  def requestId: UUID
+  def user: User
+  def giftRecipient: Option[GiftRecipient]
+  def product: ProductType
+  def paymentProvider: PaymentProvider
```

Comment:
this will force all input states that could fail, to provide the necessary fields